### PR TITLE
On Logout, fetch the CSRF token

### PR DESCRIPTION
**Pull request by ashishkumar468**
Fixes https://github.com/commons-app/apps-android-commons/pull/3150#discussion_r331743965

**Description (required)**
 * Fetch the CSRF token
 * use the CSRF token to make the logout post call
 * once both of these succeed, clear the local data

**Tests performed (required)**

Tested prodDebug on Samsung S7 API 27

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/theme/NavigationBaseActivity.java [hunk 1]
```diff
@@ -229,6 +241,59 @@ public boolean onNavigationItemSelected(@NonNull final MenuItem item) {
         }
     }
 
+    /**
+     * Fetch the CSRF token to logout
+     */
+    private void handleLogout() {
+        service = ServiceFactory.get(commonsWikiSite);
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/theme/NavigationBaseActivity.java]: Move this to a separate class. ie `LogoutClient`

File: app/src/main/java/fr/free/nrw/commons/theme/NavigationBaseActivity.java [hunk 2]
```diff
@@ -229,6 +241,59 @@ public boolean onNavigationItemSelected(@NonNull final MenuItem item) {
...
+        //First fetch the csrf token
+        if (null == progressDialog) {
+            progressDialog = new ProgressDialog(this);
+            progressDialog.setMessage(getString(R.string.please_wait));
+        }
+
+        progressDialog.show();
+        disposable.add(
+                service.getCsrfToken()
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/theme/NavigationBaseActivity.java]: Use `@Named(NAMED_COMMONS_CSRF)`

File: app/src/main/java/fr/free/nrw/commons/theme/NavigationBaseActivity.java [hunk 3]
```diff
@@ -229,6 +241,59 @@ public boolean onNavigationItemSelected(@NonNull final MenuItem item) {
...
+                                    Timber.e(t,
+                                            "CSRF token fetch failed: " + t.getLocalizedMessage());
+                                }
+                        ));
+    }
+
+    /**
+     * Make the logout post call with the CSRF token fetched
+     */
+    private void postLogoutApi(MwQueryResponse response) {
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/theme/NavigationBaseActivity.java]: Move to `LogoutClient` or `AuthClient`

**Comment by maskaravivek** [comment]
@ashishkumar468 have you opened a new PR for the same?

**Comment by ashishkumar468** [comment]
@maskaravivek Yes, this branch some how got very much away from backend-overhaul, after we recently synced backend-overhaul with master, so as the changes were small, I though making a new one would be quicker :-)

**Pull request closed without merging**